src/prerecorded/options.rs

```diff
@@ -0,0 +1,1365 @@
+    /// #     env::var("DEEPGRAM_API_KEY").expect("DEEPGRAM_API_KEY environmental variable");
+    /// #
+    /// let dg_client = Deepgram::new(&deepgram_api_key);
+    ///
+    /// # let source = UrlSource { url: AUDIO_URL };
+    /// #
+    /// let options1 = Options::builder()
+    ///     .model(Model::General)
+    ///     .multichannel_with_models([Model::Meeting, Model::Phonecall])
+    ///     .build();
+    ///
+    /// let options2 = Options::builder()
+    ///     .multichannel_with_models([Model::Meeting, Model::Phonecall])
+    ///     .build();
+    ///
+    /// let request1 = dg_client.make_prerecorded_request_builder(&source, &options1).build()?;
+    /// let request2 = dg_client.make_prerecorded_request_builder(&source, &options2).build()?;
+    ///
+    /// // Both make the same request to Deepgram with the same features
+    /// assert_eq!(request1.url(), request2.url());
+    ///
+    /// // However, they technically aren't "equal"
+    /// // This is because `options1` still remembers the model you set previously
+    /// assert_ne!(options1, options2);
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    ///
+    /// ```
+    /// # use deepgram::prerecorded::{Model, Options};
+    /// #
+    /// let options1 = Options::builder()
+    ///     .model(Model::General)
+    ///     .multichannel_with_models([Model::Meeting, Model::Phonecall])
+    ///     .multichannel(true)
+    ///     .build();
+    ///
+    /// let options2 = Options::builder()
+    ///     .model(Model::General)
+    ///     .multichannel(true)
+    ///     .build();
+    ///
+    /// assert_eq!(options1, options2);
+    /// ```
+    ///
+    /// ```
+    /// # use deepgram::prerecorded::{Model, Options};
+    /// #
+    /// let options1 = Options::builder()
+    ///     .multichannel_with_models([Model::Meeting])
+    ///     .multichannel_with_models([Model::Phonecall])
+    ///     .build();
+    ///
+    /// let options2 = Options::builder()
+    ///     .multichannel_with_models([Model::Meeting, Model::Phonecall])
+    ///     .build();
+    ///
+    /// assert_eq!(options1, options2);
+    /// ```
+    pub fn multichannel_with_models(mut self, models: impl IntoIterator<Item = Model<'a>>) -> Self {
+        if let Some(Multichannel::Enabled {
+            models: Some(old_models),
+        }) = &mut self.0.multichannel
+        {
+            // Multichannel with models already enabled
+            // Don't overwrite existing models
+            old_models.extend(models);
+        } else {
+            // Multichannel with models already enabled
+            self.0.multichannel = Some(Multichannel::Enabled {
+                models: Some(models.into_iter().collect()),
+            });
+        }
+
+        self
+    }
+
+    /// Set the maximum number of transcript alternatives to return.
+    ///
+    /// See the [Deepgram API Reference](https://developers.deepgram.com/api-reference/#alternatives-pr) for more info.
```

Comment:
you might have meant "Deepgram alternatives feature" here based on the surrounding docs